src/test/java/spoon/test/architecture/SpoonArchitectureEnforcerTest.java

```diff
@@ -206,4 +212,32 @@ public boolean matches(CtClass element) {
 			assertTrue(klass.getElements(new TypeFilter<>(CtConstructor.class)).stream().allMatch(x -> x.hasModifier(ModifierKind.PRIVATE)));
 		}
 	}
+
+	@Test
+	public void testInterfacesAreCtScannable() {
+		// contract: all interface inherited from CtElement should be visited in CtScanner
```

Comment:
the contract of this bug should be:
all non-leaf interfaces of the metamodel should be visited by CtInheritance scanner

the interfaces do not have to inherit from ctelement